runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -786,20 +788,44 @@ ClientSessionHT::purgeOldDataIfNeeded()
    {
    PORT_ACCESS_FROM_PORT(TR::Compiler->portLib);
    int64_t crtTime = j9time_current_time_millis();
+   bool incomplete;
+   bool lowMemory = false;
+   uint64_t freePhysicalMemory = compInfo->computeAndCacheFreePhysicalMemory(incomplete, 20); //check if memory is free
+   if (freePhysicalMemory != OMRPORT_MEMINFO_NOT_AVAILABLE && !incomplete)
+      {
+      if (freePhysicalMemory < TR::Options::getSafeReservePhysicalMemoryValue())
+         {
+         lowMemory = true;
+         }
+      }
    if (crtTime - _timeOfLastPurge > TIME_BETWEEN_PURGES)
       {
       // Time for a purge operation.
       // Scan the entire table and delete old elements that are not in use
       for (auto iter = _clientSessionMap.begin(); iter != _clientSessionMap.end(); ++iter)
          {
          TR_ASSERT(iter->second->getInUse() >= 0, "_inUse=%d must be positive\n", iter->second->getInUse());
-         if (iter->second->getInUse() == 0 &&
-             crtTime - iter->second->getTimeOflastAccess() > OLD_AGE)
+         if(!lowMemory)
```

Comment:
To avoid duplicating the code you have a variable `oldAge` that receives either `OLD_AGE` or `OLD_AGE_UNDER_LOW_MEMORY` depending on the situation.